File: prefixCompression/prefix.py
def prefixCompression(a,b):
    p = ""
    a1 = ""
    b1 = ""

    if len(a) > len(b):
        n = len(a)
        min = len(b)
    else:
        n = len(b)
        min = len(a)

    for i in range(n):
        if i >= min:
            if min == len(b):
                a1 += a[i]
            else:
                b1 += b[i]

        elif i <= len(b):
            if a[i] == b[i]:
                p += a[i]
            else:
                a1 += a[i]
                b1 += b[i]
            
    print(len(p),p)
    print(len(a1),a1)
    print(len(b1),b1)  

a = "kitkat"
b = "kit"
resultado = prefixCompression(a,b)

File: prefixCompression/prefix.java
package prefixCompression;

public class prefix {
    
    public static void prex(String a, String b){
        String p = "";
        String a1 = "";
        String b1 = "";
        int n = 0;
        int min = 0;
        
        if(a.length() > b.length()){
            n = a.length();
            min = b.length();
        } 
        else{
            n = b.length();
            min = a.length();
        }

        for(int i=0;i<n;i++){
            if(i >= min){
                if(min == b.length()){
                    a1 += a.charAt(i);
                }
                else{
                    b1 += b.charAt(i);
                }
            }
            else if(i <= min){
                if(a.charAt(i) == b.charAt(i)){
                    p += a.charAt(i);
                }
                else{
                    a1 += a.charAt(i);
                    b1 += b.charAt(i);
                }
            }
        }
        System.out.println(p.length());
        System.out.println(p);
        System.out.println(a1.length());
        System.out.println(a1);
        System.out.println(b1.length());
        System.out.println(b1);
    }
    
    public static void main(String[] args) {
        String a = "aaabhj";
        String b = "aaakkk";
        prex(a, b);
    }
}


